### PR TITLE
Add SBOM

### PR DESCRIPTION
**Pull request by OlleLarsson**
> [!warning]
> **This is a public repository, ensure not to disclose:**
>
> - [x] personal data beyond what is necessary for interacting with this pull request, nor
> - [x] business confidential information, such as customer names.

### What kind of PR is this?

**Required**: Mark one of the following that is applicable:

- [ ] kind/feature       
- [ ] kind/improvement   
- [ ] kind/deprecation   
- [x] kind/documentation 
- [ ] kind/clean-up      
- [ ] kind/bug           
- [ ] kind/other         

*Optional*: Mark one or more of the following that are applicable:

> [!important]
> Breaking changes should be marked `kind/admin-change` or `kind/dev-change` depending on type
> Critical security fixes should be marked with `kind/security`

- [ ] kind/admin-change 
- [ ] kind/dev-change   
- [ ] kind/security     
- [ ] [kind/adr]()      

### What does this PR do / why do we need this PR?

Add an initial SBOM based on v0.37.0

#### Checklist

- [ ] Proper commit message prefix on all commits
  
- Change checks:
  - [ ] The change is transparent
  - [ ] The change is disruptive
  - [ ] The change requires no migration steps
  - [ ] The change requires migration steps
  - [ ] The change upgrades CRDs
- Metrics checks:
  - [ ] The metrics are still exposed and present in Grafana after the change
  - [ ] The metrics names didn't change (Grafana dashboards and Prometheus alerts are not affected)
  - [ ] The metrics names did change (Grafana dashboards and Prometheus alerts were fixed)
- Logs checks:
  - [ ] The logs do not show any errors after the change
- Pod Security Policy checks:
  - [ ] Any changed pod is covered by Pod Security Admission
  - [ ] Any changed pod is covered by Gatekeeper Pod Security Policies
  - [ ] The change does not cause any pods to be blocked by Pod Security Admission or Policies
- Network Policy checks:
  - [ ] Any changed pod is covered by Network Policies
  - [ ] The change does not cause any dropped packets in the `NetworkPolicy Dashboard`
- Audit checks:
  - [ ] The change does not cause any unnecessary Kubernetes audit events
  - [ ] The change requires changes to Kubernetes audit policy
- Falco checks:
  - [ ] The change does not cause any alerts to be generated by Falco
- Bug checks:
  - [ ] The bug fix is covered by regression tests
- Config checks:
  - [ ] The schema was updated

### CODE REVIEW

**Review by cristiklein — APPROVED** [review]
Looks good for an initial SBOM. Thanks!

**Review by aarnq — COMMENTED** [review]
Is there a reason why we want to keep multiple versions of this at all times?

Else we could move it to `docs/sbom.md` and state in the document for which version it has been validated for.

Please add a top level heading "Compliant Kubernetes Apps - SBOM" or something.

**Comment by simonklb** [comment]
> Is there a reason why we want to keep multiple versions of this at all times?
> 
> Else we could move it to `docs/sbom.md` and state in the document for which version it has been validated for.
> 
> Please add a top level heading "Compliant Kubernetes Apps - SBOM" or something.

I was thinking the same, that the SBOM would reflect the current repository, so if you checked out an old tag the SBOM would reflect that version. Although if it wasn't automatically updated when the repository changed that could become confusing so maybe it's something we could add later on.

**Review by simonklb — COMMENTED** [review]
Maybe name the document 0.37.0 as main is going to drift?

**Comment by OlleLarsson** [comment]
I have pushed the new sbom,  ptal :)

File: sbom/sbom.md [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+| yq4 | 4.42.1 |  | MIT | Various contributors |  |
+| yajsv | 1.4.1 |  | MIT | Neil Pankey <npankey@gmail.com> |  |
+| pwgen | 3.4.1 |  | MIT | Theodore Ts'o tytso@alum.mit.edu |  |
+| apache2 utils | 2.4.* |  | Apache-2.0 | Apache Software Foundation |  |
+
+## Upstream Helm Charts
+
+| Name | App Version | CNCF Status | License | Copyright_owner | Comment |
+| --- | --- | --- | --- | --- | --- |
+| cert-manager | v1.12.8 | v1.12.8 | Cert-manager Authors, Linux Foundation | incubating | [Apache-2.0](https://github.com/cert-manager/cert-manager/blob/master/LICENSE) |  |
```
[the remark this answers is not in the document]
Comment (OlleLarsson) [reply sbom/sbom.md]: Thanks,  don't know how i missed that :facepalm:

**Review by cristiklein — APPROVED** [review]
Great start, thank you!

Please poke you when you merge this, because I'd like to write some public docs to it.

File: sbom/sbom.md [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+| quay.io/prometheus-operator/prometheus-config-reloader | v0.50.0 | thanos |  |  |  |  |
+| docker.io/calico/ctl | v3.26.4 | tigera-operator |  |  |  |  |
+| quay.io/tigera/operator | v1.30.9 | tigera-operator |  |  |  |  |
+| ghcr.io/aquasecurity/trivy-operator | 0.17.1 | trivy-operator |  |  |  |  |
+| ghcr.io/aquasecurity/trivy | 0.47.0 | trivy-operator |  |  |  |  |
+| docker.io/velero/velero | v1.11.1 | velero |  |  |  |  |
+| ghcr.io/elastisys/calico-accountant | 0.1.6 | calico-accountant |  |  |  |  |
+| docker.io/busybox | 1.36 | file-copier |  |  |  |  |
+| docker.io/bitnami/kubectl | 1.25 | gatekeeper-templates |  |  |  |  |
+| quay.io/prometheuscommunity/prom-label-proxy | master | grafana-label-enforcer |  |  |  |  |
```
Comment (simonklb) [inline sbom/sbom.md:189]: We probably should pin this.

**Review by simonklb — APPROVED** [review]
Nice! Looking forward to this being filled out even more and utilized in aspects such as security and compliance work!

File: sbom/sbom.md (hunk 2 above)
Comment (OlleLarsson) [reply sbom/sbom.md:189]: Yep :)

**Review by aarnq — APPROVED** [review]
LGTM! Just wondering why it isn't in the docs folder?

Also do we want to write a comment in the document for which commit of the repository it was generated for with date?

**Comment by OlleLarsson** [comment]
> LGTM! Just wondering why it isn't in the docs folder?

Sure, I don't see any problem moving it under `docs/sbom`

> Also do we want to write a comment in the document for which commit of the repository it was generated for with date?

I'm fine with adding a note in the document saying that it was based of off v0.37.0.

**Comment by OlleLarsson** [comment]
I'll merge this now. It's not perfect and there are things that need to be sorted out (like how to properly handle multiple helm charts with the same name etc) and there's probably some error in this one, but this is a start and it can only get better from here :)

**Pull request merged**